### DISCUSSION TITLE
How to do a password reset using auth as phone

### DISCUSSION THREAD

**Discussion by 2advancesolutions** [supabase/supabase#3321] [Questions] [answered]
I was looking at your documentation and for password reset it looks to only support email, If I having users sign up using phone and password and they forgot there password what api method can I use ?

**Comment by silentworks** [comment, marked as answer]
You should probably capture their email after they have signed up, as you will need this in order to communicate with them. Otherwise if you want to manage sending out the message to a phone number you can use the `supabase.auth.api.generateLink` function to generate the reset password link.

https://github.com/supabase/gotrue-js/blob/master/src/GoTrueApi.ts#L455

**Reply by 2advancesolutions** [reply]
@silentworks thank you for the quick response
